### ISSUE TITLE
Tests failing: `...running for over 60 seconds`

### ISSUE THREAD

**Issue by sectore** [DanConwayDev/ngit-cli#6]
After changes of fc3f22e (grabbed from #4) tests are failing by running into time-outs. It happened locally and on CI.

Please give #5 a try (I can't run tests on CI as an external contributor) to see if re-introducing `match_events` will make tests happy (but it seems not).

**Comment by DanConwayDev** [comment]
Thanks for raising another pull request.

TLDR; fc3f22eac2bb81823f170f61ba9d39baff76b933 which upgraded nostr  0.27 causes a number of cli output tests to fail, but running `cargo test` under a nix configuration, rather than with rustup or in ci, will cause a number of false positives.

pre nostr 0.27 upgrade, there has been a `...running for over 60 seconds` error that occur when running under the nix configuration but not in ci or under a standard rustup install. The error occurs at seemly random and inconsistent tests. When the tests are run in batches they all pass.

running `cargo test` at v0.1.0 on debian-12 vm with rustup installed works consistently.

An urgent hotfix e0f543e8adb144f6deff6ff7ea0c412c9fcac5b4 was applied between v.0.1.0 and the nosrt 0.27 upgrade which I knew would break the any tests that check the cli output. 

Interestingly running e0f543e8adb144f6deff6ff7ea0c412c9fcac5b4 or the last commit before the nost 0.27 upgrade d96c8f6687be8966a0bb25c679995fea3b9f0715 doesn't fail but produces the `...running for over 60 seconds` error consistently at the first cli output test it comes across.

The tests consistently pass on these commits when the offending lines (161 and 172 in client.rs 9fb37c8c38f55a33657c824caaabba7f72b09ea4) are commented out.

They produce the `...running for over 60 seconds` error on fc3f22eac2bb81823f170f61ba9d39baff76b933 (the nostr 0.27 upgrade) 
consistently at on the same tests.

```
     Running tests/push.rs (target/debug/deps/push-7c1008d138bb7f47)

running 6 tests
test when_branch_is_checked_out::when_branch_is_ahead::commits_issued_as_patch_event ... ok
test when_branch_is_checked_out::when_branch_is_behind::cli_prompts::cli_show_up_to_date ... ok
test when_branch_is_checked_out::when_branch_is_ahead::cli_prompts::cli_applied_1_commit has been running for over 60 seconds
test when_branch_is_checked_out::when_branch_is_up_to_date::cli_prompts::cli_show_up_to_date has been running for over 60 seconds
```
When run individually both of these tests produce the `...running for over 60 seconds` error.

I conclude from this that these tests are actually failing with changes introduced in fc3f22eac2bb81823f170f61ba9d39baff76b933.

removing those two tests show:

```
     Running tests/claim.rs (target/debug/deps/claim-f1d56c7f3e9e9260)

running 16 tests
test when_no_main_or_master_branch_return_error ... ok
test when_repo_not_previously_claimed::when_repo_relays_not_specified::cli_ouput::check_cli_output has been running for over 60 seconds
test when_repo_not_previously_claimed::when_repo_relays_not_specified::tags::relays_match_user_write_relays has been running for over 60 seconds
```
removing those two tests show:

```
     Running tests/claim.rs (target/debug/deps/claim-f1d56c7f3e9e9260)

running 14 tests
test when_no_main_or_master_branch_return_error ... ok
test when_repo_not_previously_claimed::when_repo_relays_specified_as_arguments::cli_ouput::check_cli_output has been running for over 60 seconds
test when_repo_not_previously_claimed::when_repo_relays_specified_as_arguments::sent_to_correct_relays::event_not_sent_to_fallback_relay has been running for over 60 seconds
```
There are probably more. 

There should probably be a timeout feature built into CliTester so that the test fail rather than causing the test runner run indefinitely.

**Comment by DanConwayDev** [comment]
I have a theory why the  `...running for over 60 seconds` error is occurring in nix configuration:

Rather than be related to the way the rust toolchain is installed, which was my previous assumption, it is because of https://github.com/DanConwayDev/ngit-cli/issues/1#issuecomment-1914666724. As tests are run with the reduced resources available within a nix shell, they reach some constraint and start to hang.

**Comment by DanConwayDev** [comment]
My theory was right! intermittent failing of tests under nix seems to be fixed in 45dc04b81bf7c07f249c9539a48516d4c17abdba. tests run consistently when this change is applied to v0.1.0, which I have done under dfa91edbaaaaaecf253f3b79c1a9c9425573320f.
Interestingly, I applied it to the commit before nost 0.27 upgrade, and addressed the test failures and it works most of the time but occasionally fails - a4d20eb9e087d8259b411b7bba7c6dfd13c0171c

I'll try and isolate the issues further.

**Comment by DanConwayDev** [comment]
#7 was rebased, squashed and fast forward merged in ab1450c655f7fdfc214e1556d16fc754ca684613. Hopefully it fixed the issue. so far so good.

**Comment by DanConwayDev** [comment]
on that very commit the ci test job ran successfully with `nightly` but timed out with `stable`.
https://github.com/DanConwayDev/ngit-cli/actions/runs/7752883631/job/21143150023
```
test sends_pr_and_2_patches_to_3_relays::pr_tags::pr_tags_repo_commit has been running for over 60 seconds
test sends_pr_and_2_patches_to_3_relays::pr_tags::pr_tags_title_as_name has been running for over 60 seconds
test when_commits_behind_ask_to_proceed::when_response_is_true_proceeds has been running for over 60 seconds
Error: The operation was canceled.
```

**Comment by DanConwayDev** [comment]
This appears to have been fixed via a rust-nost v0.29 upgrade.
